docs(qaas): adds qiskit-ml plugin how-to page (#6574)
scaleway/docs-content: pages/quantum-computing/how-to/use-qiskit-ml.mdx

---
title: Use Qiskit ML plugin for Quantum Machine Learning
description: Explore QML capabilities using Qiskit.
tags: quantum qiskit merlin pennylane qaas qml ml
dates:
  validation: 2026-06-08
  posted: 2026-06-08
---
import Requirements from '@macros/iam/requirements.mdx'

**[Qiskit](https://qiskit.org/)** is a comprehensive, open-source SDK for working with quantum computers. Its dedicated plugin, `qiskit-machine-learning`, [introduces foundational quantum machine learning (QML)](https://qiskit-community.github.io/qiskit-machine-learning/index.html) algorithms, neural networks, and tools to integrate quantum circuits seamlessly into classic machine learning workflows.

Scaleway offers access to the `qiskit-scaleway` provider, enabling you to run your Qiskit circuits and QML models directly on Scaleway's QaaS infrastructure with minimal configuration.

> We will build a Quantum Autoencoder using `qiskit-machine-learning` to compress and reconstruct a custom toy dataset of digital numbers (zeros and ones) on Scaleway's infrastructure. This example is freely adapted from [here](https://qiskit-community.github.io/qiskit-machine-learning/tutorials/12_quantum_autoencoder.html).

Reviewer: ```suggestion
In this practical example, we will build a Quantum Autoencoder using `qiskit-machine-learning` to compress and reconstruct a custom toy dataset of digital numbers (zeros and ones) on Scaleway's infrastructure. This example is freely adapted from [Qiskit's Quantum Autoencoder tutorial](https://qiskit-community.github.io/qiskit-machine-learning/tutorials/12_quantum_autoencoder.html).
```

## How to perform Quantum Neural Network training on Scaleway using Qiskit-ML

<Requirements />

* A Scaleway account with a valid **Project ID**

> * A Scaleway **API Key** (Secret Key)

Reviewer: ```suggestion
* A Scaleway **API key** (Secret Key)
```

* Python installed on your local machine

1. Install the `qiskit-scaleway` provider alongside the necessary machine learning and visualization packages:
    ```bash
    pip install qiskit-machine-learning qiskit-scaleway pylatexenc matplotlib
    ```

> 2. Create a helper file named `digits_dataset.py` in your working directory. This script generates a basic 5x3 grid representing handwritten "zeros" and "ones" that we will use to train our model.

Reviewer: ```suggestion
2. Create a helper file named `digits_dataset.py` in your working directory, and add the helper script shown below. This script generates a basic 5x3 grid representing handwritten "zeros" and "ones" that we will use to train our model.
```

            for r, c in one_idx():
                empty[r][c] = algorithm_globals.random.uniform(0.9, 1)
            train_images.append(empty)
            train_labels.append(1)
            if draw:
                plt.title("This is a One")
                plt.imshow(train_images[-1], vmin=0, vmax=1)
                plt.show(block=True)

        for i in range(int(num / 2)):
            empty = np.array([algorithm_globals.random.uniform(0, 0.1) for _ in range(15)]).reshape(5, 3)
            for r, c in zero_idx():
                empty[r][c] = algorithm_globals.random.uniform(0.9, 1)
            train_images.append(empty)
            train_labels.append(0)
            if draw:
                plt.imshow(train_images[-1], vmin=0, vmax=1)
                plt.title("This is a Zero")
                plt.show(block=True)

        train_images = np.array(train_images)
        # Flatten to 15-dimensional vectors for the 15 data qubits
        train_images = train_images.reshape(len(train_images), 15)

        return train_images, train_labels

    ```

3. Create the main training script. Save the following code as `qml_tutorial.py`.

>     Make sure to define your `SCW_PROJECT_ID` and `SCW_SECRET_KEY` as environment variables (or via a `.env` file) before running it. Notice how the `ScalewayProvider` is used to fetch the `EMU-AER-2L4` backend and initiate a dedicated session for the iterative `COBYLA` optimizer.

Reviewer: ```suggestion
    Make sure to define your `SCW_PROJECT_ID` and `SCW_SECRET_KEY` as environment variables (or via an `.env` file) before running it. Notice how the `ScalewayProvider` is used to fetch the `EMU-AER-2L4` backend and initiate a dedicated session for the iterative `COBYLA` optimizer.
```

            pixels[i] = sv.probabilities([i])[1]
        return pixels.reshape(5, 3)

    # Sample new images and reconstruct
    test_images, test_labels = get_dataset_digits(2, draw=False)
    for image, label in zip(test_images, test_labels):
        original_qc = feature_map.assign_parameters(image)
        original_img = extract_image(original_qc)

        param_values = np.concatenate((image, opt_result.x))
        output_qc = test_qc.assign_parameters(param_values)
        output_img = extract_image(output_qc)

        fig, (ax1, ax2) = plt.subplots(1, 2)
        ax1.imshow(original_img, vmin=0, vmax=1)
        ax1.set_title("Input Data")
        ax2.imshow(output_img, vmin=0, vmax=1)
        ax2.set_title("Output Data")
        plt.show(block=True)

    ```

4. Run the script.
    ```bash
    python ~/qml_tutorial.py
    ```

Executing this code will iteratively train the parameters of the quantum autoencoder, visually plotting the decreasing cost function in real-time. Once convergence is reached, the script will output a comparison of the newly reconstructed digits against the original input data for comparison.

> ## Understanding Session Management

Reviewer: ```suggestion
## Understanding session management
```

Because quantum machine learning algorithms often require hundreds of iterative steps (like the `COBYLA` optimization in the script above), submitting each circuit independently can introduce heavy network overhead.

> With `qiskit-scaleway`, your iterative workloads are grouped via a **session**. By explicitly calling `start_session()` on your backend and passing that `session_id` to your `ScwSampler` primitive, you guarantee that Scaleway's QaaS infrastructure processes your training loop on the same session.

Reviewer: ```suggestion
With `qiskit-scaleway`, your iterative workloads are grouped via a **session**. By explicitly calling `start_session()` on your backend and passing that `session_id` to your `ScwSampler` primitive, you guarantee that Scaleway's QaaS infrastructure processes your training loop in the same session.
```
